### ISSUE TITLE
When selecting a Sequence flow of the Label for the Sequence Flow, highlight the label/sequence

### ISSUE THREAD

**Issue by StephenOTT** [camunda/camunda-modeler#369]
__Describe the Problem__

When you select a Sequence flow the label should also be highlighted so you can quickly tell which label corresponds to the flow.

When you select a label, the sequence flow it belongs to should be highlighted so easy reference to which sequence the label belongs to.

When you have complex processes with lots of flows, the labels can be mixed up, especially when moving and resizing 

__Solution Sketch__

Show a fine, dashed line when hovering a label that links the label to the labeled element:

![image](https://user-images.githubusercontent.com/58601/60790129-09cf7000-a161-11e9-8131-9ff19075b647.png)

---

Depends on https://github.com/bpmn-io/diagram-js/pull/290

**Comment by nikku** [comment]
Sounds like a good improvement. Highlighting on `mouseover` only may be a snappier solution, too.

**Comment by StephenOTT** [comment]
Yes that would be great.  👍

**Comment by pinussilvestrus** [comment]
- [ ] highlight label(s) when selecting sequence flow
- [ ] highlight sequence flow when selecting label
- [ ] + mouseover

**Comment by nikku** [comment]
This needs additional thought and work to be a proper feature that does not confuse users.

We postponed the implementation in diagram-js (https://github.com/bpmn-io/diagram-js/pull/290) to a later point in time.

**Comment by pinussilvestrus** [comment]
Additional approach: showing a dashed line which indicates the reference between label and sequence flow (or any other element?)

![image](https://user-images.githubusercontent.com/9433996/58860806-36e5ba00-86ad-11e9-8f7b-a18e0471fb27.png)

**Comment by nikku** [comment]
I think that would be an amazing option :rocket: :fire:

**Comment by jarekdanielak** [comment]
Let's decide on a solution and get this issue ready to be worked on. 

From the solutions proposed in the comments above, I think a dashed line between an element and its label is the most fitted for this. It solves the problem and fits our design/UX style.

I suggest this dashed line only appears when the label is far enough from its parent, see:

![Image](https://github.com/user-attachments/assets/1d45ef84-5323-4ace-b647-157e8914e304)

**Comment by barmac** [comment]
Sounds good to me. This should resolve the problem without cluttering the view. We can also reconsider if it's not good enough.

**Comment by nikku** [comment]
Let's test "dashed line if not too far" is actually annoying. I think it is not and would personally go for the most simplest solution (always have a dashed line) and iterate from there.

**Comment by barmac** [comment]
To Nico's point, we want to display the line only for hover/selection.

**Comment by jarekdanielak** [comment]
OK I assumed the line will be always displayed, but if it's only on hover/selection, I agree we can ignore the "distance check".

**Comment by lmbateman** [comment]
I think the dashed line is a good solution and don't mind always showing it on hover/selection, but in Jarek's first two screenshots, there's not a lot of room for the line, and I'm wondering if a single dash will convey the relationship. (But maybe it will look better than I expect.)

**Comment by jarekdanielak** [comment]
I assume this connection should be just a visual cue and we don't want to create an actual `bpmn:Association`, right?

**Comment by barmac** [comment]
> I assume this connection should be just a visual cue and we don't want to create an actual `bpmn:Association`, right?

Correct, it's just a visual helper similar to the selection outline.

**Comment by jarekdanielak** [comment]
My prototype: https://github.com/bpmn-io/bpmn-js/pull/2328

I have to park this for now, hopefully to be picked up next spring cleaning.

**Comment by jarekdanielak** [comment]
Fixed upstream in https://github.com/bpmn-io/bpmn-js/pull/2328
